### PR TITLE
Migration CRA to VITE

### PR DESCRIPTION
**Pull request by shivam-sharma7**
CRA (create-react-app) is no longer maintained. Vite is a build tool that aims to provide a faster and leaner development experience for modern web projects

closes: #995

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by shivam-sharma7** [comment]
I have tested it locally and build got success and vite config also working fine.

**Comment by shivam-sharma7** [comment]
@johnaohara check once again

**Review by johnaohara — CHANGES_REQUESTED** [review]
We also want to;

- remove `horreum-web/src/setupProxy.js` 
- we now have 2 different index.html's,  `horreum-web/index.html` & `horreum-web/public/index.html`.  We need to remove the old file `horreum-web/public/index.html` and ensure that the quarkus filter redirects to the correct index.html (https://github.com/Hyperfoil/Horreum/blob/846c9687b58950c646bed6e9189bf587da7d3460/horreum-backend/src/main/java/io/hyperfoil/tools/horreum/server/RouteFilter.java)

File: horreum-web/vite.config.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        // this sets a default port to 3000  
+        port: 3000, 
+
+        proxy: {
+            '/api': {
+                target: 'http://localhost:8080',
+                changeOrigin: true,
+                rewrite: (path) => path.replace(/^\/api/, '')
+            }
+        }
```
[the remark this answers is not in the document]
Comment (shivam-sharma7) [reply horreum-web/vite.config.ts:21]: done 👍

**Comment by shivam-sharma7** [comment]
> We also want to;
> 
> * remove `horreum-web/src/setupProxy.js`

done

> * we now have 2 different index.html's,  `horreum-web/index.html` & `horreum-web/public/index.html`.  We need to remove the old file `horreum-web/public/index.html` and ensure that the quarkus filter redirects to the correct index.html (https://github.com/Hyperfoil/Horreum/blob/846c9687b58950c646bed6e9189bf587da7d3460/horreum-backend/src/main/java/io/hyperfoil/tools/horreum/server/RouteFilter.java)

WDYT? for current changes RouteFilter.. do we have to realy change?

File: horreum-backend/src/main/java/io/hyperfoil/tools/horreum/server/RouteFilter.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
  */
 @ApplicationScoped
 public class RouteFilter {
-   private static final String[] PATH_PREFIXES = { "/api/", "/connect", "/dev" };
+   private static final String[] PATH_PREFIXES = { "/q/", "/api/", "/@" };
```
Comment (shivam-sharma7) [inline horreum-backend/src/main/java/io/hyperfoil/tools/horreum/server/RouteFilter.java:15]:
```suggestion
   private static final String[] PATH_PREFIXES = { "/q/", "/api/", "/@" };
```

let's try this

**Comment by johnaohara** [comment]
I have removed this from the 0.11 milestone.  It is not critical for the 0.11 release and we can make this change in a point release after 0.11.0

**Comment by shivam-sharma7** [comment]
@johnaohara if you thing its good then please aprrove this and resolve the conversation

**Review by johnaohara — APPROVED** [review]
@shivam-sharma7 this looks good to me now, please can you rebase on top of master and we can get this merged, many thanks

**Comment by johnaohara** [comment]
@shivam-sharma7 do you see the node.js dev server closed when the horreum dev process is stopped? I had an orphaned node.js process still listening on port 3000 after I stopped the dev mode

**Comment by shivam-sharma7** [comment]
> @shivam-sharma7 do you see the node.js dev server closed when the horreum dev process is stopped? I had an orphaned node.js process still listening on port 3000 after I stopped the dev mode

I will take a look

**Comment by johnaohara** [comment]
@shivam-sharma7 how did you generate `package-lock.json` for this PR? When i run `npm i` in `horreum-web` now a lot of dependencies are removed and is causing build issues with vite on windows

**Comment by shivam-sharma7** [comment]
@johnaohara Here I got build got success and in order to  generate `package-lock.json` I deleted old one and I run `npm i` in `horreum-web`
```java
on-tests\0.12-SNAPSHOT\horreum-integration-tests-0.12-SNAPSHOT-tests.jar
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Horreum 0.12-SNAPSHOT:
[INFO]
[INFO] Horreum ............................................ SUCCESS [  1.400 s]
[INFO] horreum-api ........................................ SUCCESS [ 14.719 s]
[INFO] Horreum Client ..................................... SUCCESS [  1.233 s]
[INFO] Horreum infra - common ............................. SUCCESS [  0.533 s]
[INFO] Horreum Dev Services - Parent ...................... SUCCESS [  0.032 s]
[INFO] Horreum Dev Services - Runtime ..................... SUCCESS [  5.204 s]
[INFO] Horreum Dev Services - Deployment .................. SUCCESS [  5.614 s]
[INFO] Horreum Backend .................................... SUCCESS [02:55 min]
[INFO] Horreum Integration Tests .......................... SUCCESS [01:01 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  04:33 min
[INFO] Finished at: 2024-01-25T13:55:31+05:30
```

**Pull request merged**